src/main/java/com/faforever/client/replay/ReplayService.java

```diff
@@ -503,6 +507,14 @@ private void runFafReplayFile(Path path) throws IOException {
       mapName = parseMapFolderName(rawReplayBytes);
     }
 
+    // For map generator games the map name is "None" because replay server gets map name by from DB based on filename
+    // from replay data, and DB does not contain generated maps.
+    if (StringUtils.equalsIgnoreCase(mapName, "None")) {
+      String maybeMapGen = parseMapFolderName(rawReplayBytes);
+      if (mapGeneratorService.isGeneratedMap(maybeMapGen)) {
+        mapName = maybeMapGen;
+      }
+    }
```

Comment:
Could we write a unit test for that?

Reply:
Do you have some actual specification that determines when mapName is supposed to be null, empty or "None"? Otherwise the test will just enforce arbitrary implementation.

Reply:
Sheikah added a test